test/sqllogictest/filter-pushdown.slt

```diff
@@ -0,0 +1,137 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+# Verify filter pushdown information for various temporal filters.
+# For straightforward temporal filters like these, every column mentioned in the filter
+# should be present in the pushdown list.
+
+statement ok
+CREATE TABLE events (
+    content text,
+    insert_ms numeric,
+    delete_ms numeric
+);
+
+query T multiline
+EXPLAIN WITH(mfp_pushdown)
+SELECT count(*)
+FROM events
+WHERE mz_now() >= insert_ms
+  AND mz_now() < delete_ms;
+----
+Explained Query:
+  Return
+    Union
+      Get l0
+      Map (0)
+        Union
+          Negate
+            Project ()
+              Get l0
+          Constant
+            - ()
+  With
+    cte l0 =
+      Reduce aggregates=[count(*)]
+        Project ()
+          Filter (mz_now() < numeric_to_mz_timestamp(#2)) AND (mz_now() >= numeric_to_mz_timestamp(#1))
+            Get materialize.public.events
+
+Source materialize.public.events
+  filter=((mz_now() >= numeric_to_mz_timestamp(#1)) AND (mz_now() < numeric_to_mz_timestamp(#2)))
+  pushdown=(#1, #2)
```

Comment:
What is `(#1, #2)` meant to signify here? It seems to be referring to `filter` conjuncts rather than columns, which is somewhat confusing because the `#<number>` syntax always represents a column reference elsewhere...

Reply:
It does actually refer to columns! This line can be read as "we can push down a filter on columns 1 and 2". (You'll notice that other examples have `pushdown=(#1)`, because their filters do not reference column 2 in a way that allows us to use knowledge about the possible values in column 2 to reason about the possible outputs of the filter.